### PR TITLE
kalphite thru wyrms

### PR DESCRIPTION
**Pull request by wyattos**

### CODE REVIEW

File: src/monsters/bosses/KalphiteQueen.ts [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+export const KqConsumablesTable = new LootTable()
+	.addItem('Monkfish', 3, 1)
+	.addItem('Shark', 2, 1)
+	.addItem('Dark crab', 2, 1)
+	.addItem('Saradomin brew(4)', 1, 1)
+	.addItem('Prayer potion(4)', 2, 1)
+	.addItem('Super restore(4)', 1, 1)
+	.addItem('Super combat potion(2)', 1, 1)
+	.addItem('Ranging potion(3)', 1, 1)
+	.addItem('Superantipoison(2)', 1, 1);
```
Comment (gc) [inline src/monsters/bosses/KalphiteQueen.ts]: Fix name

File: src/monsters/bosses/KalphiteQueen.ts [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+	.tertiary(100, 'Clue scroll (elite)')
+	.tertiary(128, 'Kq head')
+	.tertiary(2000, 'Jar of sand')
+	.tertiary(3000, 'Kalphite princess');
+
+export default new SimpleMonster({
+	id: 128,
+	name: 'Kalphite Queen',
+	table: KalphiteQueenTable,
+	aliases: ['kalphite queen', 'kq', 'Kq', 'KQ']
```
Comment (gc) [inline src/monsters/bosses/KalphiteQueen.ts]: aliases are case insensitive and only lowercase, so should be just 2 here

File: src/monsters/nieve/Kraken.ts [hunk 3]
```diff
@@ -0,0 +1,66 @@
+import LootTable from '../../structures/LootTable';
+import SimpleMonster from '../../structures/SimpleMonster';
+import { GemTable } from '../subtables/RareDropTable';
+
+export const KrakenTable = new LootTable(128)
+
+	/*Weapons and armour */
```
Comment (gc) [inline src/monsters/nieve/Kraken.ts]: spacing of all of these

File: src/monsters/nieve/LizardmanShaman.ts [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+	.tertiary(200, 'Clue scroll (hard)')
+	.tertiary(400, 'Long bone')
+	.tertiary(1200, 'Clue scroll (elite)')
+	.tertiary(5013, 'Curved bone');
+
+export default new SimpleMonster({
+	id: 6766,
+	name: 'Lizardman Shaman',
+	table: LizardmanShamanTable,
+	aliases: ['lizardman shaman', 'shaman', 'Shaman']
```
Comment (gc) [inline src/monsters/nieve/LizardmanShaman.ts]: case insensitive

File: src/monsters/nieve/LocustRider.ts [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+	.tertiary(200, 'Clue scroll (hard)')
+	.tertiary(400, 'Long bone')
+	.tertiary(1200, 'Clue scroll (elite)')
+	.tertiary(5013, 'Curved bone');
+
+export default new SimpleMonster({
+	id: 795,
+	name: 'Locust Rider',
+	table: LocustRiderTable,
+	aliases: ['Locust', 'locust rider']
```
Comment (gc) [inline src/monsters/nieve/LocustRider.ts]: aliases again

File: src/monsters/nieve/SkeletalWyvern.ts [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+	.addItem('Runite crossbow (u)', 1, 2)
+	.addItem('Ranarr seed', 3, 2)
+	.addItem('Snapdragon seed', 1, 2)
+
+	/* Rdt */
+	.add(RareDropTable, undefined, 3)
+
+	/* Tertiary */
+	.tertiary(350, 'Clue scroll (elite)')
+	.tertiary(10000, 'Draconic visage');
```
Comment (gc) [inline src/monsters/nieve/SkeletalWyvern.ts]: should be `10_000`, if longer than 4 digits, have a `_` in it for easier reading

File: src/monsters/nieve/SmokeDevil.ts [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+	.addItem('Rune dagger', 1, 3)
+	.addItem('Air battlestaff', 1, 3)
+	.addItem("Black d'hide vamb", 1, 3)
+	.addItem('Fire battlestaff', 1, 3)
+	.addItem('Mithril plateskirt', 1, 2)
+	.addItem('Rune full helm', 1, 2)
+	.addItem('Rune chainbody', 1, 2)
+	.addItem('Rune chainbody', 1, 1)
+	.oneIn(512, 'Occult necklace')
+	.oneIn(32768, 'Dragon chainbody')
```
Comment (gc) [inline src/monsters/nieve/SmokeDevil.ts]: `_` here

File: src/monsters/nieve/SteelDragon.ts [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+	.addItem('Curry', 2, 1)
+
+	/* Rdt */
+	.add(RareDropTable, undefined, 4)
+	.add(GemTable, undefined, 4)
+
+	/* Tertiary */
+	.tertiary(64, 'Clue scroll (hard)')
+	.tertiary(500, 'Clue scroll (elite)')
+	.tertiary(10000, 'Draconic visage')
```
Comment (gc) [inline src/monsters/nieve/SteelDragon.ts]: `_`

File: src/monsters/nieve/SteelDragon.ts [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+
+	/* Catacombs tertiary*/
```
Comment (gc) [inline src/monsters/nieve/SteelDragon.ts]: remove this and all catacomb stuff in this PR

File: src/monsters/nieve/SmokeDevil.ts [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+	/* Weapons and armor */
+	.addItem('Adamant battleaxe', 1, 3)
+	.addItem('Rune dagger', 1, 3)
+	.addItem('Air battlestaff', 1, 3)
+	.addItem("Black d'hide vamb", 1, 3)
+	.addItem('Fire battlestaff', 1, 3)
+	.addItem('Mithril plateskirt', 1, 2)
+	.addItem('Rune full helm', 1, 2)
+	.addItem('Rune chainbody', 1, 2)
+	.addItem('Rune chainbody', 1, 1)
```
Comment (Alexsuperfly) [inline src/monsters/nieve/SmokeDevil.ts]:
```suggestion
	.addItem("Red d'hide body", 1, 1)
```

File: src/monsters/nieve/SmokeDevil.ts [hunk 11]
```diff
@@ -0,0 +1,66 @@
...
+	/* Rdt */
+	.add(RareDropTable, 4)
+	.add(GemTable, 4)
+
+	/* Tertiary */
+	.tertiary(128, 'Clue scroll (hard)')
+	.tertiary(750, 'Clue scroll (elite)');
+
+export default new SimpleMonster({
+	id: 425,
```
Comment (Alexsuperfly) [inline src/monsters/nieve/SmokeDevil.ts]:
```suggestion
	id: 498,
```

File: src/monsters/nieve/SpiritualWarrior.ts [hunk 12]
```diff
@@ -0,0 +1,42 @@
...
+	.addItem('Mithril axe', 1, 8)
+	.addItem('Steel sword', 1, 7)
+	.addItem('Mithril mace', 1, 7)
+	.addItem('Black warhammer', 1, 5)
+	.addItem('Iron sword', 1, 3)
+	.addItem('Black dagger', 1, 2)
+	.addItem('Adamant battleaxe', 1, 2)
+	.addItem('Adamant 2h sword', 1, 1)
+	.addItem('Rune longsword', 1, 1)
+	.addItem('Rune longsword', 1, 1)
```
Comment (Alexsuperfly) [inline src/monsters/nieve/SpiritualWarrior.ts]:
```suggestion
	.addItem('Rune halberd', 1, 1)
```

File: src/monsters/nieve/SteelDragon.ts [hunk 13]
```diff
@@ -0,0 +1,51 @@
...
+
+	/* Runes and ammunition */
+	.addItem('Rune javelin', 7, 20)
+	.addItem('Blood rune', 20, 19)
+	.addItem('Runite bolts', [2, 12], 6)
+	.addItem('Soul rune', 5, 5)
+
+	/* Coins */
+	.addItem('Coins', 470, 17)
+	.addItem('Coins', 650, 5)
```
Comment (Alexsuperfly) [inline src/monsters/nieve/SteelDragon.ts]: We've been assuming MM2 completion everywhere else do that there

File: src/monsters/nieve/Turoth.ts [hunk 14]
```diff
@@ -0,0 +1,54 @@
...
+	.add(HerbDropTable, 1, 15)
+	.add([[HerbDropTable], [HerbDropTable]], 1, 10)
+	.add([[HerbDropTable], [HerbDropTable], [HerbDropTable]], 1, 6);
+
+export const TurothTable = new LootTable(128)
+	.every('Bones')
+
+	/* Weapons and armour */
+	.addItem('Steel platelegs', 1, 7)
+	.addItem('Steel platelegs', 1, 3)
```
Comment (Alexsuperfly) [inline src/monsters/nieve/Turoth.ts]:
```suggestion
	.addItem('Mithril axe', 1, 3)
```

File: src/monsters/nieve/Wyrm.ts [hunk 15]
```diff
@@ -0,0 +1,57 @@
...
+	.addItem('Dragon dagger', 1, 1)
+
+	/* Runes */
+	.addItem('Fire rune', 200, 10)
+	.addItem('Earth rune', [75, 150], 10)
+	.addItem('Soul rune', [15, 20], 5)
+	.addItem('Blood rune', [25, 30], 5)
+
+	/* Herbs */
+	.add(HerbDropTable, 2, 4)
```
Comment (Alexsuperfly) [inline src/monsters/nieve/Wyrm.ts]: `Two random herbs are dropped when this herb drop table is rolled`

File: src/monsters/special/TzKalZuk.ts [hunk 16]
```diff
@@ -0,0 +1,11 @@
+import LootTable from '../../structures/LootTable';
+import SimpleMonster from '../../structures/SimpleMonster';
+
+export const TzTokJadTable = new LootTable().every('Infernal cape').oneIn(75, 'Jal-nib-rek');
```
Comment (Alexsuperfly) [inline src/monsters/special/TzKalZuk.ts]:
```suggestion
export const TzKalZukTable = new LootTable().every('Infernal cape').oneIn(75, 'Jal-nib-rek');
```

File: src/monsters/special/TzKalZuk.ts [hunk 17]
```diff
@@ -0,0 +1,11 @@
...
+
+export default new SimpleMonster({
+	id: 7706,
+	name: 'TzKalZuk',
+	table: TzTokJadTable,
```
Comment (Alexsuperfly) [inline src/monsters/special/TzKalZuk.ts]:
```suggestion
	table: TzKalZukTable,
```

File: src/monsters/special/TzKalZuk.ts [hunk 18]
```diff
@@ -0,0 +1,11 @@
...
+	aliases: ['tzkalzuk', 'zuk']
```
Comment (Alexsuperfly) [inline src/monsters/special/TzKalZuk.ts]:
```suggestion
	aliases: ['tzkalzuk', 'zuk', 'inferno']
```

File: src/monsters/special/TzTokJad.ts [hunk 19]
```diff
@@ -0,0 +1,11 @@
+import LootTable from '../../structures/LootTable';
+import SimpleMonster from '../../structures/SimpleMonster';
+
+export const TzTokJadTable = new LootTable().every('Fire cape').oneIn(100, 'Tzrek-jad');
+
+export default new SimpleMonster({
+	id: 3127,
+	name: 'TzTokJad',
+	table: TzTokJadTable,
+	aliases: ['tztokjad', 'jad']
```
Comment (Alexsuperfly) [inline src/monsters/special/TzTokJad.ts]:
```suggestion
	aliases: ['tztokjad', 'jad', 'fight cave']
```

File: src/monsters/subtables/WyvernHerbTable.ts [hunk 20]
```diff
@@ -0,0 +1,11 @@
...
+
+const WyvernHerbTable = new LootTable()
+	.addItem('Grimy kwuarm', 2, 260)
+	.addItem('Grimy cadantine', 2, 208)
+	.addItem('Grimy dwarf weed', 2, 208)
+	.addItem('Grimy ranarr weed', 1, 185)
+	.addItem('Grimy torstol', 1, 185)
+	.addItem('Grimy lantadyme', 2, 155);
+
+export default WyvernHerbTable;
```
Comment (Alexsuperfly) [inline src/monsters/subtables/WyvernHerbTable.ts]: this doesnt seem used why is it included

File: src/monsters/nieve/Wyrm.ts [hunk 21]
```diff
@@ -31,7 +31,7 @@ export const WyrmTable = new LootTable(76)
 	.addItem('Blood rune', [25, 30], 5)
 
 	/* Herbs */
-	.add(HerbDropTable, 2, 4)
+	.add([HerbDropTable, HerbDropTable], 2, 4)
```
Comment (Alexsuperfly) [inline src/monsters/nieve/Wyrm.ts]:
```suggestion
	.add([HerbDropTable, HerbDropTable], 1, 4)
```

File: src/monsters/nieve/Wyrm.ts [hunk 22]
```diff
@@ -0,0 +1,57 @@
...
+	.addItem('Dragon dagger', 1, 1)
+
+	/* Runes */
+	.addItem('Fire rune', 200, 10)
+	.addItem('Earth rune', [75, 150], 10)
+	.addItem('Soul rune', [15, 20], 5)
+	.addItem('Blood rune', [25, 30], 5)
+
+	/* Herbs */
+	.add([HerbDropTable, HerbDropTable], 1, 4)
```
Comment (Alexsuperfly) [inline src/monsters/nieve/Wyrm.ts]:
```suggestion
	.add([[HerbDropTable], [HerbDropTable]], 1, 4)
```

File: src/monsters/nieve/Nechryael.ts [hunk 23]
```diff
@@ -0,0 +1,51 @@
...
+	/* Runes*/
+	.addItem('Chaos rune', 37, 8)
+	.addItem('Death rune', 5, 6)
+	.addItem('Death rune', 10, 6)
+	.addItem('Law rune', [25, 35], 5)
+	.addItem('Blood rune', [15, 20], 4)
+
+	/* Seeds */
+	.addItem('Limpwurt seed', 1, 6)
+	.add(RareSeedTable, 2, 18)
```
Comment (Alexsuperfly) [inline src/monsters/nieve/Nechryael.ts]: pretty sure this is two separate rolls not 2 of the same roll

File: src/monsters/subtables/RareSeedTable.ts [hunk 24]
```diff
@@ -12,7 +12,7 @@ const RareSeedTable = new LootTable()
 	.addItem('Snapdragon seed', 1, 10)
 	.addItem('Cadantine seed', 1, 7)
 	.addItem('Lantadyme seed', 1, 5)
-	.addItem('Snape grass seed', 3, 4)
+	.addItem('Snape grass seed', 1, 4)
```
Comment (Alexsuperfly) [inline src/monsters/subtables/RareSeedTable.ts]: revert this

**Pull request closed without merging**